.github/deployment/scripts/protocol-cli.rs

```diff
@@ -368,50 +414,60 @@ fn update_env_file(path: &str, updates: &HashMap<String, String>) -> Result<()>
         .with_context(|| format!("Failed to write environment file: {}", path))
 }
 
-/// Executes any command with proper environment setup
-async fn run_command(command: &str, args: &[&str]) -> Result<String> {
-    // Output the command being executed
-    info(&format!("Executing: {} {}", command, args.join(" ")));
-
-    let output = tokio::process::Command::new(command)
-        .args(args)
-        .env_clear()  // Clear existing environment
-        .envs(env::vars())  // Add all current environment variables
-        .output()
-        .await?;
-
-    if !output.status.success() {
-        let stderr = String::from_utf8_lossy(&output.stderr);
-        bail!("Command failed: {} {}\nError: {}", command, args.join(" "), stderr);
-    }
-
-    Ok(String::from_utf8_lossy(&output.stdout).to_string())
-}
+/// Executes any command with retry logic
+async fn run_command(command: &str, args: &[&str], max_retries: u32, retry_errors: &[&str]) -> Result<String> {
+    let mut last_error = None;
+    
+    for attempt in 1..=max_retries {
+        info(&format!("Executing: {} {}", command, args.join(" ")));
+
+        let output = tokio::process::Command::new(command)
+            .args(args)
+            .env_clear()
+            .envs(env::vars())
+            .output()
+            .await?;
+
+        if !output.status.success() {
+            let stderr = String::from_utf8_lossy(&output.stderr);
+            let error = anyhow!("Command failed: {} {}\nError: {}", command, args.join(" "), stderr);
+            let error_msg = error.to_string();
+            
+            // Check if error matches any retry patterns (case-insensitive)
+            let error_msg_lower = error_msg.to_lowercase();
```

Comment:
I would likely create a helper that did: `should_retry(retry_errors, error_msg)` that hides all the lowercasing and iteration logic.
you could also make `retry_errors` a struct and have that helper there, then you don't need to pass `retry_errors` to the method. Up to you.